src/core/Track.cpp

```diff
@@ -1070,12 +1113,48 @@ void TrackContentObjectView::mouseReleaseEvent( QMouseEvent * me )
 	{
 		setSelected( !isSelected() );
 	}
-
-	if( m_action == Move || m_action == Resize || m_action == ResizeLeft )
+	else if( m_action == Move || m_action == Resize || m_action == ResizeLeft )
 	{
 		// TODO: Fix m_tco->setJournalling() consistency
 		m_tco->setJournalling( true );
 	}
+	else if( m_action == Split )
+	{
+		SampleTCO * leftTCO = dynamic_cast<SampleTCO*>( m_tco );
+
+		if (leftTCO)
+		{
+			leftTCO->setMarkerEnabled( false );
+
+			const int relativePixelPos = me->pos().x();
+			const float ppt = m_trackView->trackContainerView()->pixelsPerTact();
+			MidiTime splitPos = relativePixelPos * MidiTime::ticksPerTact() / ppt;
+
+			if ( !unquantizedModHeld(me) )
+			{
+				splitPos = quantizeMarkerPos( splitPos, me->modifiers() & Qt::ShiftModifier );
+			}
+
+			splitPos += m_initialTCOPos;
+
+			//Don't split if we slid off the TCO or if we're on the clip's start/end
+			//Cutting at exactly the start/end position would create a zero length
+			//clip (bad), and a clip the same length as the original one (pointless).
+			if ( splitPos > m_initialTCOPos && splitPos < m_initialTCOEnd )
+			{
+				leftTCO->copy();
+				SampleTCO * rightTCO = new SampleTCO ( leftTCO->getTrack() );
+				rightTCO->paste();
+
+				leftTCO->changeLength( splitPos - m_initialTCOPos );
+
+				rightTCO->movePosition( splitPos );
+				rightTCO->changeLength( m_initialTCOEnd - splitPos );
+				rightTCO->setStartTimeOffset( leftTCO->startTimeOffset() - leftTCO->length() );
+			}
```

Comment:
Can we move the TCO-type-specific splitting logic into its own class for each TCO type? I think it will make it easy to extend this feature to other TCO types.

Reply:
Also, do you really need to use the `copy()` and `paste()` calls? It will silently change the content of the clipboard.

Reply:
> move the TCO-type-specific splitting logic

Good idea.

> do you really need to use the copy() and paste() calls

Probably not, but I have no idea how else to do it. I tried some things on my own but only ever managed to make either the left or the right clip. Since BaraMGB had solved it I just took that existing code and used it.

Either way I can't work on this again until November (at earliest). If you're able to push changes to this and want to apply the above in the meantime, feel free to do so.

Reply:
> do you really need to use the copy() and paste() calls?

Fixed in latest commit.